File: v1-sdd-only/server/src/main/java/com/ebbinghaus/vocab/domain/entity/StudyPlan.java
package com.ebbinghaus.vocab.domain.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

@Data
@TableName("study_plan")
public class StudyPlan {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long userId;

    private Long bookId;

    private Integer planWordCount;

    private Integer reviewMultiplier;

    private Integer dailyReviewCount;

    private Integer dailyTotalCount;
}

File: v2-polished/client/src/api/studyPlan.ts
import request from './request'

export interface StudyPlanData {
  id: number; userId: number; bookId: number | null;
  planWordCount: number; reviewMultiplier: number;
  dailyReviewCount: number; dailyTotalCount: number;
}

export function getStudyPlan() {
  return request.get<{ code: number; message: string; data: StudyPlanData }>('/study-plan')
}

export function updateStudyPlan(p: { bookId?: number; planWordCount?: number; reviewMultiplier?: number }) {
  return request.put<{ code: number; message: string; data: StudyPlanData }>('/study-plan', p)
}
